app/code/Magento/PageCache/Model/Varnish/VclGenerator.php

```diff
@@ -67,7 +75,8 @@ public function __construct(
         $accessList,
         $gracePeriod,
         $sslOffloadedHeader,
-        $designExceptions = []
+        $designExceptions = [],
+        ?RequestInterface $request = null
```

Comment:
Let's disallow to pass null instead of required dependencies (default null value will still be applicable in this case)
```suggestion
        RequestInterface $request = null
```